spec/system/consumer/authentication_spec.rb

```diff
@@ -156,6 +156,34 @@
       end
     end
 
+    describe "Logging in from the private shop page" do
+      let(:distributor) { create(:distributor_enterprise, require_login: true) }
+      let!(:order_cycle) {
+        create(:simple_order_cycle, distributors: [distributor],
+                                    coordinator: create(:distributor_enterprise))
+      }
+      before do
+        visit enterprise_shop_path(distributor)
+      end
+
+      it "clicking Login triggers the login modal" do
+        within "#shop-tabs" do
+          find("a", text: "login").click
+        end
+        expect(page).to have_selector("a.active", text: "Login")
+        expect(page).to have_button("Login")
+      end
+
+      # needs to be changed once #8860 is addressed
+      it "clicking Signup triggers login modal and opens the signup tab" do
+        within "#shop-tabs" do
+          find("a", text: "signup").click
+        end
+        expect(page).to have_selector("a.active", text: "Login")
+        expect(page).to have_button "Login"
+      end
```

Comment:
Great find! There's a better way of describing this test case though:
```suggestion
      it "clicking Signup triggers login modal and opens the signup tab" do
        pending "#8860 - it currently opens the login tab
        within "#shop-tabs" do
          find("a", text: "signup").click
        end
        expect(page).to have_selector("a.active", text: "Sign up")
        expect(page).to have_button "Sign up now"
      end
```

Reply:
Thanks for sharing that! It's much more informative and readable than just using `xit`.